src/runner/browser-set.js

```diff
@@ -32,10 +32,18 @@ export default class BrowserSet extends EventEmitter {
         if (bc.idle || bc.closed || !bc.ready)
             return;
 
+        var idlePromise  = promisifyEvent(bc, 'idle');
+        var closePromise = promisifyEvent(bc, 'closed');
+
         await Promise.race([
-            promisifyEvent(bc, 'idle'),
-            promisifyEvent(bc, 'closed')
+            idlePromise,
+            closePromise
         ]);
+
+        // We must delete both listeners from browser connection after
+        // execution one of them to avoid overflow subscribers limit.
```

Comment:
```
        // We must delete both listeners from the browser connection after
        // one of them is executed to avoid exceeding the subscribers limit.
```